mock/py/mockbuild/backend.py

```diff
@@ -699,7 +699,6 @@ def get_command(mode):
                 # * it fails
                 # * installSrpmDeps does nothing
                 # * or we run out of dynamic_buildrequires_max_loops tries
-                packages_before = self.buildroot.all_chroot_packages()
```

Comment:
Yes, we can drop this (no need to access the RPM db by `rpm -qa`), but we
need a replacement here.  Most likely you need to maintain a `set()` of
real build requirements across the iteration...  in each step you can get
the list from `installSrpmDeps` with small enhancements.